src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -3323,6 +3347,93 @@ mgp_error mgp_graph_delete_edge(struct mgp_graph *graph, mgp_edge *edge) {
+      std::visit([graph, id](auto *impl) { return impl->FindVertex(id, graph->view); }, graph->impl);
+  if (maybe_vertex) {
+    return std::visit(memgraph::utils::Overloaded{
+                          [memory, graph, maybe_vertex](memgraph::query::DbAccessor *) {
+                            return NewRawMgpObject<mgp_vertex>(memory, *maybe_vertex, graph);
+                          },
+                          [memory, graph, maybe_vertex](memgraph::query::SubgraphDbAccessor *impl) {
+                            return NewRawMgpObject<mgp_vertex>(
+                                memory, memgraph::query::SubgraphVertexAccessor(*maybe_vertex, impl->getGraph()),
+                                graph);
+                          }},
+                      graph->impl);
+  }
+  return nullptr;
+}
+
+void WrapTextSearch(std::vector<memgraph::storage::Gid> vertex_ids, std::string error_msg, mgp_graph *graph,
+                    mgp_memory *memory, mgp_map **result) {
+  if (const auto err = mgp_map_make_empty(memory, result); err != mgp_error::MGP_ERROR_NO_ERROR) {
+    throw std::logic_error("Retrieving text search results failed during creation of a mgp_map");
+  }
+
+  mgp_list *search_results{};
+  if (const auto err = mgp_list_make_empty(vertex_ids.size(), memory, &search_results);
+      err != mgp_error::MGP_ERROR_NO_ERROR) {
+    throw std::logic_error("Retrieving text search results failed during creation of a mgp_list");
+  }
+
+  for (const auto &vertex_id : vertex_ids) {
+    mgp_value *vertex;
+    if (const auto err = mgp_value_make_vertex(GetVertexByGid(graph, vertex_id, memory), &vertex);
+        err != mgp_error::MGP_ERROR_NO_ERROR) {
+      throw std::logic_error("Retrieving text search results failed during creation of a vertex mgp_value");
+    }
+    if (const auto err = mgp_list_append(search_results, vertex); err != mgp_error::MGP_ERROR_NO_ERROR) {
+      throw std::logic_error(
+          "Retrieving text search results failed during insertion of the mgp_value into the result list");
+    }
+  }
+
+  mgp_value *search_results_value;
+  if (const auto err = mgp_value_make_list(search_results, &search_results_value);
+      err != mgp_error::MGP_ERROR_NO_ERROR) {
+    throw std::logic_error("Retrieving text search results failed during creation of a map mgp_value");
+  }
+  mgp_value *error_msg_value;
+  if (const auto err = mgp_value_make_string(error_msg.data(), memory, &error_msg_value);
+      err != mgp_error::MGP_ERROR_NO_ERROR) {
+    throw std::logic_error("Retrieving text search results failed during creation of a map mgp_value");
+  }
+
+  if (const auto err = mgp_map_insert(*result, "search_results", search_results_value);
+      err != mgp_error::MGP_ERROR_NO_ERROR) {
+    throw std::logic_error("Retrieving text search results failed during insertion into mgp_map");
+  }
+  if (const auto err = mgp_map_insert(*result, "error_msg", error_msg_value); err != mgp_error::MGP_ERROR_NO_ERROR) {
```

Comment:
In the above comment I described why `std::optional` is not necessary.
That said, while we \*can\* use it, I strongly hold that we should refrain from using `std::optional` because that would cause the scope of this PR to explode:
1. Map values are `mgp_value` by type. We’d have to extend `mgp_value` with optional types since they aren’t supported now.
2. This would imply new API methods like `mgp_value_make_optional_#type`, `mgp_value_is_optional_#type`, `mgp_type_optional_#type`, and `mgp_value_get_optional_#type`
3. Analogous changes for the APIs in `_mgp.hpp` and `mgp.hpp`.

Reply:
no no, I don't want optional mgp_type... Never mind, I meant something else, should be good